src/taipy/core/scenario/scenario.py

```diff
@@ -135,19 +135,26 @@ def pipelines(self, pipelines: Union[List[PipelineId], List[Pipeline]]):
 
     @property
     def tasks(self) -> Dict[str, Task]:
-        task_manager = _TaskManagerFactory._build_manager()
+        return self._get_tasks()
+
+    def _get_tasks(self):
         tasks = {}
-        list_tasks = [pipeline.tasks for pipeline in self.pipelines.values()]
-        for task in list_tasks:
-            for k, v in task.items():
-                t = task_manager._get(v, v)
-                if not isinstance(t, Task):
-                    raise NonExistingTask(v)
-                tasks[k] = v
+        list_dict_tasks = [pipeline.tasks for pipeline in self.pipelines.values()]
+        for dict_task in list_dict_tasks:
+            for task_config_id, task in dict_task.items():
+                if task_config_id in tasks.keys() and task not in tasks[task_config_id]:
+                    tasks[task_config_id].append(task)
+                else:
+                    tasks[task_config_id] = [task]
```

Comment:
```suggestion
                    if task_config_id in tasks:
                        tasks[task_config_id].append(task)
                    else:
                        tasks[task_config_id] = [task]  
```

Reply:
Do we need to check `task not in tasks[task_config_id]`? Because, if not, there can be the same task in the list of tasks because the same task can belong to multiple pipeline

Reply:
You are right. We should have something like that: 

```
if task_config_id in tasks: 
    if task not in tasks[task_config_id]:
        tasks[task_config_id].append(task)
else:
    tasks[task_config_id] = [task]
```

By the way, the function could return a `Dict[str, Set[Task]]`. Am I right?

Reply:
Yes and no 😃 we are also checking if the Task is already existing in the dictionary[task_config_id], if yes, we won't append the Task => no duplication similar to that of a Set object. But in the end, the value is still returning a List, just with no duplication

Reply:
Let's imagine the following scenario: 
```
A -> t1 ---> B -> t2  -> C -> t3 -> D
             |
              --> t2' -> E
```
with 3 pipelines in this order (t1, t2), (t1, t2'), and (t2, t3). 
t2 and t2' are two different tasks, but they have the same config_id.

Here is the value of 
```
list_dict_tasks = [
{cfg_1: t1, cfg_2: t2}
{cfg_1: t1, cfg_2: t2'}
{cfg_2: t2, cfg_3: t3}
]
```

Here are the successive values of tasks.

tasks = {cfg_1: [t1]}
tasks = {cfg_1: [t1], cfg_2: [t2]}
tasks = {cfg_1: [t1], cfg_2: [t2]}
tasks = {cfg_1: [t1], cfg_2: [t2, t2']}
tasks = {cfg_1: [t1], cfg_2: [t2]}
tasks = {cfg_1: [t1], cfg_2: [t2], cfg_3: [t3]}

At the end t2' is missing.

Can you implement such test to be sure?

I believe the following solves the issue:
```suggestion
if task_config_id in tasks: 
    if task not in tasks[task_config_id]:
        tasks[task_config_id].append(task)
else:
    tasks[task_config_id] = [task]
```